tests/Unit/Core/Module/ModuleManagerTest.php

```diff
@@ -230,11 +232,19 @@ private function getModuleDataProviderMock(): ModuleDataProvider
     {
         $moduleDataProvider = $this->createMock(ModuleDataProvider::class);
 
+        $isInstalledCallback = new class() {
+            private $count = 0;
+
+            public function isInstalled($name)
+            {
+                return $name === ModuleManagerTest::INSTALLED_MODULE_NAME
+                    || ($name === ModuleManagerTest::INSTALLED_THEN_UNINSTALLED_MODULE_NAME
+                        && ++$this->count < 3);
+            }
+        };
+
         $moduleDataProvider->method('isInstalled')
-            ->willReturnMap([
-                [self::INSTALLED_MODULE_NAME, true],
-                [self::UNINSTALLED_MODULE_NAME, false],
-            ])
+            ->willReturnCallback([$isInstalledCallback, 'isInstalled'])
```

Comment:
Honestly, I don't understand this part. What is this "Installed, then uninstalled" scenario about?

Reply:
Yeah, it's a bit tricky 😅 When you reset a module, there is 2 ways: using the `reset` function or using `uninstall` then `install`. With the second way, we need to be sure that the module is installed before calling `uninstall` and uninstalled before calling `install`. This callback returns `true` twice and then `false` to simulate the expected behavior

Reply:
@atomiix, now it makes sense :D What do you think about adding your comment to the code?